xml_converter/generators/code_generator.py

```diff
@@ -359,13 +388,58 @@ def build_example(self, type: str, applies_to: List[str], xml_field: str, exampl
 
         return example
 
+    ############################################################################
+    # Generate Node Types
+    #
+    # This will output code for a single category of nodes.
+    ############################################################################
+    def generate_node_types(self, metadata: Any, attributenames: Dict[str,str],page: str) -> Tuple[List[str],List[str]]:
+        
+        field_rows: List[str] = []
+        cpptypes: List[str] = []
```

Comment:
This does not really store cpptypes but instead stores the dependencies that need to be included (minus the `.hpp` extention).

Additionally you might want to change this from a `List[str]` to a `Set[str]`